rclcpp/include/rclcpp/memory_strategy.hpp

```diff
@@ -64,9 +64,11 @@ class RCLCPP_PUBLIC MemoryStrategy
   virtual void clear_handles() = 0;
   virtual void remove_null_handles(rcl_wait_set_t * wait_set) = 0;
 
-  virtual void add_guard_condition(const rcl_guard_condition_t * guard_condition) = 0;
+  virtual void
+  add_guard_condition(const rclcpp::GuardCondition * guard_condition) = 0;
```

Comment:
This is another place where we could take a reference instead of a pointer.

Reply:
Done in ccc78c5